samples/bluetooth/peripheral_hids_keyboard/src/main.c

```diff
@@ -752,12 +785,39 @@ static void button_shift_changed(bool down)
 	}
 }
 
+static void num_comp_reply(struct bt_conn *conn, bool accept)
+{
+	if (accept) {
+		bt_conn_auth_passkey_confirm(conn);
+		printk("Numeric Match, conn %p\n", conn);
+	} else {
+		bt_conn_auth_cancel(conn);
+		printk("Numeric Reject, conn %p\n", conn);
+	}
+}
+
 static void button_changed(u32_t button_state, u32_t has_changed)
 {
-	if (has_changed & KEY_TEXT_MASK) {
+	u32_t buttons = button_state & has_changed;
+
+	if (auth_conn) {
+		if (buttons & KEY_PAIRING_ACCEPT) {
+			num_comp_reply(auth_conn, true);
+		}
+
+		if (buttons & KEY_PAIRING_REJECT) {
+			num_comp_reply(auth_conn, false);
+		}
+
+		auth_conn = NULL;
+
+		return;
+	}
+
+	if (buttons & KEY_TEXT_MASK) {
```

Comment:
Is
```has_changed & KEY_TEXT_MASK```
equal to:
```buttons & KEY_TEXT_MASK```?

I am not sure about this. You want to call ```button_text_changed``` function when buttons are either pressed or released.

Reply:
Oh I was to fast with changed, reverted.